classifier/disk.go

```diff
@@ -72,3 +95,58 @@ func checkRangeCapacity(min, max, count bmh.Capacity) bool {
 	}
 	return true
 }
+
+// checkDisk it filter outs the disks from bmh disk array as per hardware details 
+func checkDisk(pattern []hwcc.DiskSelector, disks []bmh.Storage) ([]bmh.Storage, bool) {
+	var diskNew []bmh.Storage
+
+	for _, pattern := range pattern {
+		matched := false
+		for _, disk := range disks {
+			validatedPattern := validatePattern(disk.HCTL)
+			validatedExpectedPattern := validateExpectedPattern(pattern.HCTL)
+
+			if validatedExpectedPattern == validatedPattern && pattern.Rotational == disk.Rotational {
+				matched = true
+				diskNew = append(diskNew, disk)
+			}
+		}
+
+		if !matched {
+			log.Info("Disk Pattern",
+				"pattern", pattern,
+				"ok", false,
+			)
+
+			return diskNew, false
+		}
+	}
+
+	return diskNew, true
+}
+
+// validatePattern finds out the disk with the pattern provided in hardware profile
+func validatePattern(HCTL string) string {
+
+	if HCTL == "0:0:0:0" {
+		return "0:0:N:0"
+
+	} else {
+		res1 := strings.Split(HCTL, ":")
+		for index, st := range res1 {
+			value, _ := strconv.Atoi(st)
+			if value > 0 {
+				res1[index] = "N"
+			}
+		}
+		return strings.Join(res1, ":")
```

Comment:
change var name res1 to resp every other places.

Reply:
DONE